lib/type-helpers/mixin-type.helper.ts

```diff
@@ -0,0 +1,44 @@
+import { Type } from '@nestjs/common';
+import { Field } from '../decorators';
+import { getFieldsAndDecoratorForType } from '../schema-builder/utils/get-fields-and-decorator.util';
+import {
+  ClassDecoratorFactory,
+  inheritValidationMetadata,
+  inheritTransformationMetadata,
+} from './type-helpers.utils';
+
+export function MixinType<A, B>(
```

Comment:
What about using `IntersectionType` instead? We use `mixin` keyword for a different functionality already

Reply:
It is more like `Union` than `Intersection`.

Reply:
By the definition of "intersection" in general, you're right. However, in TypeScript, Intersection types are defined as follows:

> An intersection type combines multiple types into one. This allows you to add together existing types to get a single type that has all the features you need. For example, Person & Serializable & Loggable is a Person and Serializable and Loggable. That means an object of this type will have all members of all three types. You will mostly see intersection types used for mixins and other concepts that don’t fit in the classic object-oriented mold. (There are a lot of these in JavaScript!) Here’s a simple example that shows how to create a mixin:

Ref https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types

Reply:
OK. Since TypeScript says `A & B` is intersection type then we use `IntersectionType` as it's name